src/Resources/contao/library/Contao/Folder.php

```diff
@@ -319,27 +319,88 @@ public function copyTo($strNewName)
 
 	/**
 	 * Protect the folder by removing the .public file
+	 *
+	 * @deprecated use denyPublicAccess() instead
 	 */
 	public function protect()
 	{
-		if (file_exists(TL_ROOT . '/' . $this->strFolder . '/.public'))
-		{
-			$objFile = new \File($this->strFolder . '/.public');
-			$objFile->delete();
-		}
+		@trigger_error(
+			'Using protect() has been deprecated and will no longer work in Contao 5.0. Use denyPublicAccess() instead.',
+			E_USER_DEPRECATED
+		);
+
+		$this->denyPublicAccess();
 	}
 
 	/**
 	 * Unprotect the folder by adding a .public file
+	 *
+	 * @deprecated use allowPublicAccess() instead
 	 */
 	public function unprotect()
 	{
-		if (!file_exists(TL_ROOT . '/' . $this->strFolder . '/.public'))
-		{
-			\File::putContent($this->strFolder . '/.public', '');
-		}
+		@trigger_error(
+			'Using unprotect() has been deprecated and will no longer work in Contao 5.0. Use allowPublicAccess() instead.',
+			E_USER_DEPRECATED
+		);
+
+		$this->allowPublicAccess();
 	}
 
+	/**
+	 * Check if the folder (or any parent) is public by checking if the .public file is set.
+	 *
+	 * @param bool $includeParents
+	 *
+	 * @return bool
+	 */
+    public function isPublic(bool $includeParents = true): bool
```

Comment:
This method should be named `isUnprotected` (as discussed in Mumble on October 18th).